src/juju/apiHooks.ts

```diff
@@ -76,53 +76,59 @@ export const useListSecrets = (
   const dispatch = useAppDispatch();
   const modelUUID = useSelector(getModelUUIDFromList(modelName, userName));
   const wsControllerURL = useAppSelector(getWSControllerURL);
-
-  const fetchSecrets = useCallback(
-    (connection?: ConnectionWithFacades | null, error?: string | null) => {
-      if (error && wsControllerURL) {
-        dispatch(
-          jujuActions.setSecretsErrors({
-            modelUUID,
-            errors: error,
-            wsControllerURL,
-          }),
-        );
-        return;
-      }
-      if (!connection || !wsControllerURL) {
-        return;
-      }
-      dispatch(jujuActions.secretsLoading({ modelUUID, wsControllerURL }));
-      connection?.facades.secrets
-        ?.listSecrets({
-          ...(filter ? { filter } : {}),
-          "show-secrets": showSecrets,
-          // The type provided by Juju's schema.json has the filter as required.
-        } as ListSecretsArgs)
-        .then((secrets) => {
-          dispatch(
-            jujuActions.updateSecrets({
-              modelUUID,
-              secrets: secrets?.results ?? [],
-              wsControllerURL,
-            }),
-          );
-          return;
-        })
-        .catch((error) => {
+  const modelConnectionCallback = useModelConnectionCallback(modelUUID);
+  return useCallback(() => {
+    modelConnectionCallback(
```

Comment:
This hook has been changed to return a callback to fetch the secrets. The rest of the changes are just indentation changes.